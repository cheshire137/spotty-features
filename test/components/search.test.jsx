import fetchMock from 'fetch-mock'
import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

import Search from '../../src/components/search.jsx'

function props(unauthorized) {
  return {
    token: '123abc',
    unauthorized
  }
}

describe('Search', () => {
  let component = null
  let wasUnauthorized = false

  const unauthorized = () => {
    wasUnauthorized = true
  }

  beforeEach(() => {
    component = <Search {...props(unauthorized)} />
  })

  afterEach(fetchMock.restore)

  test('matches snapshot', () => {
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
