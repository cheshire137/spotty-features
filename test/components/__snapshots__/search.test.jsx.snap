exports[`Search matches snapshot 1`] = `
<div>
  <form
    onSubmit={[Function]}>
    <h3
      className="title is-3">
      Step 1: Seed your playlist
    </h3>
    <div
      className="control">
      <label
        className="label"
        htmlFor="seed">
        Recommendations based on this 
        song
        :
      </label>
      <div
        className="results-container">
        <input
          autoComplete="off"
          autoFocus={true}
          className="input is-large"
          id="seed"
          onChange={[Function]}
          onKeyUp={[Function]}
          placeholder="Search songs on Spotify"
          type="text"
          value="" />
        <ul
          className="results"
          style={
            Object {
              "display": "none",
            }
          } />
      </div>
    </div>
    <div
      className="control">
      <label
        className="radio is-large">
        <input
          checked={true}
          name="seed-type"
          onChange={[Function]}
          type="radio"
          value="track" />
        Songs
      </label>
      <label
        className="radio is-large">
        <input
          checked={false}
          name="seed-type"
          onChange={[Function]}
          type="radio"
          value="artist" />
        Artists
      </label>
    </div>
  </form>
</div>
`;
